components/TxConfirmationDialog.qml

```diff
@@ -0,0 +1,454 @@
+            PropertyChanges { target: backButton; visible: true; primary: false }
+            PropertyChanges { target: confirmButton; visible: true }
+            StateChangeScript { script: confirmButton.forceActiveFocus() }
+        }, State {
+            // error message being displayed, show only back button
+            name: "error";
+            when: errorText.text !==""
+            PropertyChanges { target: dialogTitle; text: "Error" }
+            PropertyChanges { target: errorText; visible: true }
+            PropertyChanges { target: txAmountText; visible: false }
+            PropertyChanges { target: txAmountBusyIndicator; visible: false }
+            PropertyChanges { target: txFiatAmountText; visible: false }
+            PropertyChanges { target: txDetails; visible: false }
+            PropertyChanges { target: bottom; visible: true }
+            PropertyChanges { target: bottomMessage; visible: false }
+            PropertyChanges { target: buttons; visible: true }
+            PropertyChanges { target: backButton; visible: true; primary: true }
+            PropertyChanges { target: confirmButton; visible: false }
+            StateChangeScript { script: backButton.forceActiveFocus() }
+        }, State {
+            // creating or sending transaction, show tx details and don't show any button
+            name: "bottomText";
+            when: errorText.text =="" && bottomText.text !==""
+            PropertyChanges { target: errorText; visible: false }
+            PropertyChanges { target: txAmountText; visible: root.transactionAmount !== "(all)" || (root.transactionAmount === "(all)" && currentWallet.isHwBacked() === true) }
+            PropertyChanges { target: txAmountBusyIndicator; visible: !txAmountText.visible }
+            PropertyChanges { target: txFiatAmountText; visible: txAmountText.visible && persistentSettings.fiatPriceEnabled && root.transactionAmount !== "(all)" }
+            PropertyChanges { target: txDetails; visible: true }
+            PropertyChanges { target: bottom; visible: true }
+            PropertyChanges { target: bottomMessage; visible: true }
+            PropertyChanges { target: buttons; visible: false }
+        }
+    ]
+
+    // same signals as Dialog has
+    signal accepted()
+    signal rejected()
+    signal closeCallback();
+
+    function open(description, amount, address, priority) {
+        if (description) {
+            root.transactionDescription = description;  
+        }
+        if (amount) {
+            root.transactionAmount = amount;  
+        }
+        if (address) {
+            root.transactionAddress = address;  
+        }
+        if (priority) {
+            root.transactionPriority = priority;  
+        }
+
+        root.visible = true;
+        
+        //clean previous error message
+        errorText.text = "";        
+    }
+    
+    function close() {
+        root.visible = false;
+        closeCallback();
```

Comment:
The closeCallbacks in both TxConfirmationDialog.qml and SuccessfulTxDialog.qml seem unused and can be removed. (Unless I missed something)